testng-core/src/main/java/org/testng/internal/DataProviderMethodRemovable.java

```diff
@@ -0,0 +1,22 @@
+package org.testng.internal;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import org.testng.IDataProviderMethod;
+import org.testng.annotations.IDataProviderAnnotation;
+
+/** Represents an @{@link org.testng.annotations.DataProvider} annotated method. */
+class DataProviderMethodRemovable extends DataProviderMethod {
```

Comment:
We're clearing up DataProviderMethod only if it's of `DataProviderMethodRemovable` type. That's because some tests (for example `test.github1490.VerifyDataProviderListener#testInstanceBasedDataProviderInformation`) rely on dataprovider instance and method being saved after the test run.

Reply:
What if we did the following ?

* Add an extra default method called `boolean isDynamic()` to `org.testng.IDataProviderMethod` (Mainly because this is a publicly exposed interface)
* Toggle the setting of this getter of `org.testng.IDataProviderMethod` by adding an overloaded constructor into `org.testng.internal.DataProviderMethod` which sets up the boolean to `true|false`

Then we would not need this extra class. I am basically trying to check if we can avoid this class, especially when all its trying to do is to differentiate a dynamic data provider vs a normal data provider.

Reply:
We can do the same thing via a boolean that is toggled and thus we can get rid of this additional class.

Reply:
done